api/pull.go

```diff
@@ -221,6 +221,10 @@ func (api *PullAPI) UninstallFilter(id rpc.ID) bool {
 	api.mux.Lock()
 	defer api.mux.Unlock()
 
+	return api.uninstallFilter(id)
+}
+
+func (api *PullAPI) uninstallFilter(id rpc.ID) bool {
```

Comment:
Is it safe to also do the same with `addFilter`? Meaning, to add the lock in the exported functions that make calls to `addFilter`.

Reply:
We could, but the idea behind locking and not locking is not really related to exported/unexported methods. Adding a lock to exported method that is named the same as unexported method without a lock is just a pattern where you have to use both. If we would remove a lock from addFilter and move it to other methods that call it we would lock bigger section that doesn't need to be blocked (it calls for latest block etc, so make a bigger section blocking which is not ideal). So I would say we leave it as it is and if we realise we would need finer management of locks we can move then.